### PR TITLE
feat: improve items on activities screen

### LINKED ISSUE #558: add call to action buttons in the pending sections
cf.

### PR DESCRIPTION
**Pull request by rhbrunetto**
## Changes

- Adds ActivityTile to ui;
- Fixes minor UI issues on Activities Screen;
- Refactors pending activity items to use ActivityTile.

<details><summary>Screenshot</summary>

Simulator Screen Shot - iPhone SE (3rd generation) - 2023-01-18 at 16 43 51

Simulator Screen Shot - iPhone SE (3rd generation) - 2023-01-18 at 16 43 49

</details> 

## Related issues

Fixes #558 

## Checklist

- [x] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [x] Screenshots/video added.
- [ ] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

**Comment by ookami-kb** [comment]
@rhbrunetto please resolve conflicts

File: packages/cryptoplease/lib/features/payment_request/src/bl/repository.dart [hunk 1]
```diff
@@ -49,6 +49,10 @@ class PaymentRequestRepository {
   Future<void> save(PaymentRequest payment) =>
       _db.into(_db.paymentRequestRows).insertOnConflictUpdate(payment.toRow());
 
+  Future<void> delete(PaymentRequest payment) =>
+      (_db.delete(_db.paymentRequestRows)..whereSamePrimaryKey(payment.toRow()))
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/payment_request/src/bl/repository.dart:53]: Doesn't make sense to convert the model to a row just for comparing keys later. You can filter by the primary key directly.

File: packages/cryptoplease/lib/features/outgoing_direct_payments/src/bl/odp_request_repository.dart [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    final query = _db.select(_db.oDPRequestRows)..where((p) => p.id.equals(id));
+
+    return query.watchSingleOrNull().map((row) => row?.toModel(_tokens));
+  }
+
+  Future<void> save(ODPRequest request) =>
+      _db.into(_db.oDPRequestRows).insertOnConflictUpdate(request.toDto());
+
+  Future<void> delete(ODPRequest request) =>
+      (_db.delete(_db.oDPRequestRows)..whereSamePrimaryKey(request.toDto()))
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/outgoing_direct_payments/src/bl/odp_request_repository.dart:38]: Same here.

File: packages/cryptoplease/lib/features/activities/src/components/odp_request_tile.dart [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  Widget build(BuildContext context) => ActivityTile(
+        title: context.l10n.odpRequestTitle(
+          activity.data.amount.format(DeviceLocale.localeOf(context)),
+          activity.data.label ?? '',
+        ),
+        subtitle: context.formatDate(activity.created),
+        icon: Assets.icons.outgoing.svg(),
+        actions: [
+          CpButton(
+            text: 'Pay',
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/activities/src/components/odp_request_tile.dart:28]: Should be in l10n.

File: packages/cryptoplease/lib/features/activities/src/components/odp_request_tile.dart [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            minWidth: 120,
+            size: CpButtonSize.micro,
+            onPressed: () => context.confirmODPRequest(activity.data),
+          ),
+          CpButton(
+            text: 'Decline',
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/activities/src/components/odp_request_tile.dart:34]: Should be in l10n.

File: packages/cryptoplease/lib/features/activities/src/components/odp_tile.dart [hunk 5]
```diff
@@ -6,37 +6,21 @@ import '../../../../core/presentation/format_date.dart';
...
-            )
-          ],
-        ),
-        subtitle: Text(
-          context.formatDate(activity.created),
-          style: subtitleStyle,
-        ),
-        leading: Assets.icons.outgoing.svg(),
+  Widget build(BuildContext context) => ActivityTile(
+        title: 'Sent directly',
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/activities/src/components/odp_tile.dart:19]: Should be in l10n.

File: packages/cryptoplease/lib/features/activities/src/components/oksp_tile.dart [hunk 6]
```diff
@@ -6,37 +6,30 @@ import '../../../../core/presentation/format_date.dart';
...
-          ],
-        ),
-        subtitle: Text(
-          context.formatDate(activity.created),
-          style: subtitleStyle,
-        ),
-        leading: Assets.icons.outgoing.svg(),
-        onTap: () => context.router.navigate(OSKPRoute(id: activity.id)),
+  Widget build(BuildContext context) => ActivityTile(
+        title: 'Sent via link',
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/activities/src/components/oksp_tile.dart:20]: Should be in l10n.

File: packages/cryptoplease/lib/features/activities/src/components/payment_request_tile.dart [hunk 7]
```diff
@@ -81,19 +79,30 @@ class _PaymentRequestTileState extends State<PaymentRequestTile> {
...
-              title: Text(title(), style: titleStyle),
-              subtitle: Text(
-                context.formatDate(data.created),
-                style: subtitleStyle,
-              ),
+              title: title(),
+              subtitle: context.formatDate(data.created),
+              actions: [
+                CpButton(
+                  text: 'Remind me',
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/activities/src/components/payment_request_tile.dart:92]: Should be in l10n. Also, the text is misleading. It should rather be "Open" or smth like this.

File: packages/cryptoplease/lib/features/activities/src/components/payment_request_tile.dart [hunk 8]
```diff
@@ -81,19 +79,30 @@ class _PaymentRequestTileState extends State<PaymentRequestTile> {
...
+                  minWidth: 120,
+                  size: CpButtonSize.micro,
+                  onPressed: () =>
+                      context.navigateTo(LinkDetailsFlowRoute(id: data.id)),
+                ),
+                CpButton(
+                  text: 'Cancel',
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/activities/src/components/payment_request_tile.dart:99]: Should be in l10n.

File: packages/cryptoplease/lib/features/outgoing_direct_payments/src/bl/odp_request.dart [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:solana/solana.dart';
+
+import '../../../../core/amount.dart';
+
+part 'odp_request.freezed.dart';
+
+@freezed
+class ODPRequest with _$ODPRequest {
+  const factory ODPRequest({
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/outgoing_direct_payments/src/bl/odp_request.dart:10]: What is the difference between the existing `PaymentRequest` and this one?

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: packages/cryptoplease/lib/features/outgoing_direct_payments/src/bl/odp_request.dart (hunk 9 above)
Comment (rhbrunetto) [reply packages/cryptoplease/lib/features/outgoing_direct_payments/src/bl/odp_request.dart:10]: `PaymentRequest` is the entity created when you create a payment request on the main wallet screen, generating a Solana Pay Link to be shared. The `ODPRequest` is created when you open a Solana Pay Link, allowing the app to track the payment status when you send money.

Comment (ookami-kb) [reply packages/cryptoplease/lib/features/outgoing_direct_payments/src/bl/odp_request.dart:10]: But in that case, it's just an `OutgoingDirectPayment`.

Comment (rhbrunetto) [reply packages/cryptoplease/lib/features/outgoing_direct_payments/src/bl/odp_request.dart]: Removed in the last update, as discussed.

**Comment by ookami-kb** [comment]
@rhbrunetto please resolve conflicts.

**Review by ookami-kb — CHANGES_REQUESTED** [review]
The behavior is pretty inconsistent:



- First item: details page is opened by clicking both on the item itself and on "Share Links" button.
- Second item: details page is opened only by clicking on "View" button.
- Third link: no buttons, details page is opened by clicking on the item.

**Pull request merged**